test/topology_custom/test_mv_delete_partitions.py

```diff
@@ -0,0 +1,71 @@
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, key, values, concurrency):
+    logger.info(f"Starting writes for key {key} with concurrency {concurrency}")
+    async def do_inserts(i: int):
+        # Insert all rows with keys with the same modulo as i
+        insert_stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+        while i < values:
+            await cql.run_async(insert_stmt, [key, i, key + i])
+            # await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {i}, {key + i})")
+            i += concurrency
+    tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+
+    await asyncio.gather(*tasks)
+    logger.info(f"Finished writes for key {key} with concurrency {concurrency}")
+
+async def wait_for_views(cql, mvs_count, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        logger.info(f"Views built: {done[0][0]}")
+        if done[0][0] == node_count * mvs_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+# This test reproduces issue #12379
+# The test scenario is the following:
+# 1. Create a table with 30000 rows in 30 partitions (1000 rows per partition)
+# 3. Create and wait for 30 MVs
+# 4. Delete all partitions, causing 30*30*1000 = 0.9M view updates
+# If the test doesn't cause a crash due to OOM in the step 4, it passes
+@pytest.mark.asyncio
+@skip_mode('debug', "-m option doesn't work in debug mode")
+async def test_delete_partition_rows_from_table_with_mv(manager: ManagerClient) -> None:
+    node_count=2
+    await manager.servers_add(node_count, cmdline=['-m', '300MB'])
+    cql = manager.get_cql()
+    mv_name_pref = "mv_cf_view"
+    mvs_count = 30
+    partitions_count = 30
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': 1}}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v int, primary key (key, c))")
+    for key in range(partitions_count):
+        await insert_with_concurrency(cql, key, 1000, 100)
+
+    for i in range(mvs_count):
+        await cql.run_async(f"CREATE MATERIALIZED VIEW ks.{mv_name_pref}_{i} AS SELECT * FROM ks.tab "
+                        "WHERE c IS NOT NULL and key IS NOT NULL and v IS NOT NULL PRIMARY KEY (v, c, key) ")
+
+    await wait_for_views(cql, mvs_count, node_count)
+
+    keys = ", ".join([str(key) for key in range(partitions_count)])
+    logger.info(f"Deleting rows for keys {keys}")
+    await cql.run_async(f"DELETE FROM ks.tab WHERE KEY IN ({keys})", timeout=300)
```

Comment:
Is the timeout=300 setting enough? Don't we also need "USING TIMEOUT"? Please try to check what is the difference between the two.

Reply:
The entire test takes less than 100 seconds, so 300 should probably also be enough. In the end I didn't change the default `write_request_timeout_in_ms` with "USING TIMEOUT" because the default is also 300 seconds. Hopefully the timeouts won't be relevant for the future iterations